airmozilla/popcorn/renderer.py

```diff
@@ -0,0 +1,101 @@
+import os
+import tempfile
+import time
+from uuid import uuid4
+
+import boto
+from boto.s3.key import Key
+from funfactory.urlresolvers import reverse
+from popcoder.popcoder import process_json
+
+from django.db import transaction
+from django.conf import settings
+from django.utils import timezone
+
+from airmozilla.base.utils import build_absolute_url, prepare_vidly_video_url
+from airmozilla.popcorn.models import PopcornEdit
+from airmozilla.main.models import Event, VidlySubmission
+from airmozilla.manage import vidly
+from airmozilla.uploads.models import Upload
+
+
+@transaction.atomic
+def render_edit(edit_id):
+    edit = PopcornEdit.objects.get(id=edit_id)
+    event = edit.event
+
+    filename = '%s.webm' % edit.id
+    filepath = os.path.join(tempfile.gettempdir(), filename)
+    if not (os.path.isfile(filepath) and
+            os.stat(filepath)[os.stat.st_size] > 0):
+        edit.status = PopcornEdit.STATUS_PROCESSING
+        edit.save()
+        process_json(edit.data, out=filepath)
+
+    upload_file = uuid4().hex[:13]
```

Comment:
The variable `upload_file` isn't great. 
Is it a name or a path? Or is it an instance of a Upload class? See what I mean?